spring-integration-http/src/main/java/org/springframework/integration/http/inbound/HttpRequestHandlingEndpointSupport.java

```diff
@@ -638,6 +637,9 @@ private void postProcessRequest(HttpServletRequest request) {
 	@SuppressWarnings({"unchecked", "rawtypes"})
 	private Object extractRequestBody(ServletServerHttpRequest request) throws IOException {
 		MediaType contentType = request.getHeaders().getContentType();
+		if (contentType == null) {
+			contentType = MediaType.APPLICATION_OCTET_STREAM;
```

Comment:
Looks like we are correct here finally :smile:  !
`AbstractMessageConverterMethodArgumentResolver`:
```
if (contentType == null) {
	noContentType = true;
	contentType = MediaType.APPLICATION_OCTET_STREAM;
}
```
Something similar is in the `ConsumeMediaTypeExpression`:
```
MediaType contentType = StringUtils.hasLength(request.getContentType()) ?
						MediaType.parseMediaType(request.getContentType()) :
						MediaType.APPLICATION_OCTET_STREAM;
```

So, LGTM - merging...